unittests/allplatformstests.py

```diff
@@ -4884,6 +4884,26 @@ def output_name(name, type_):
                 with self.subTest(key='{}.{}'.format(data_type, file)):
                     self.assertEqual(res[data_type][file], details)
 
+    @skip_if_not_language('rust')
+    @unittest.skipIf(not shutil.which('rustfmt'), 'Test requires rustfmt')
+    def test_rustfmt(self) -> None:
+        if self.backend is not Backend.ninja:
+            raise unittest.SkipTest('Rust is only supported with ninja currently')
+        try:
+            with tempfile.TemporaryDirectory() as tmpdir:
+                testdir = self.copy_srcdir(os.path.join(self.rust_test_dir, '9 unit tests'))
+                self.init(testdir)
+                with self.assertRaises(subprocess.CalledProcessError) as cm:
+                    self.build('rustfmt-check')
+
+                self.build('rustfmt')
+                self.build('rustfmt-check')
+        except PermissionError:
+            # When run under Windows CI, something (virus scanner?)
+            # holds on to the git files so cleaning up the dir
+            # fails sometimes.
+            pass
```

Comment:
You shouldn't need the with tempfile.TemproraryDirectory or the `try/except` with the `copy_scrdir`, it takes care of both of those issues.

Reply:
Oh, sorry - I misunderstood your previous review.